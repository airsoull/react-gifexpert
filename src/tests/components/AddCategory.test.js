import React from 'react';

const { shallow } = require('enzyme');
const { default: AddCategory } = require('../../components/AddCategory');

describe('test for AddCategory component', () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test('should show component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should change input value', () => {
    const input = wrapper.find('input');
    const value = 'foo-bar';

    input.simulate('change', { target: { value } });
    expect(wrapper.find('input').props().value).toBe(value);
  });

  test('should not post information on submit form', () => {
    const preventDefault = jest.fn();
    wrapper.find('form').simulate('submit', { preventDefault });

    expect(preventDefault).toHaveBeenCalledTimes(1);
    expect(setCategories).not.toHaveBeenCalled();
  });

  test('should call setCategories and clean inputValue when submit form', () => {
    const value = 'foo-bar';
    const input = wrapper.find('input');
    const preventDefault = jest.fn();

    input.simulate('change', { target: { value } });
    wrapper.find('form').simulate('submit', { preventDefault });

    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(input.props().value).toBe('');
  });
});
