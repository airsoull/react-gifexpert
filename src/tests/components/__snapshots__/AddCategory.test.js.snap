// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for AddCategory component should show component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Search yours GIFs..."
    type="text"
    value=""
  />
</form>
`;
